contrib/aws/aws-sdk-go-v2/aws/aws.go

```diff
@@ -101,6 +123,324 @@ func (mw *traceMiddleware) startTraceMiddleware(stack *middleware.Stack) error {
 	}), middleware.After)
 }
 
+func extractResourceNameFromParams(requestInput middleware.InitializeInput, awsService string) (resourceNameKey string, resourceNameValue string) {
+	switch awsService {
+	case "SQS":
+		resourceNameKey, resourceNameValue = extractQueueName(requestInput)
+	case "S3":
+		resourceNameKey, resourceNameValue = extractBucketName(requestInput)
+	case "SNS":
+		resourceNameKey, resourceNameValue = extractTopicName(requestInput)
+	case "DynamoDB":
+		resourceNameKey, resourceNameValue = extractTableName(requestInput)
+	case "Kinesis":
+		resourceNameKey, resourceNameValue = extractStreamName(requestInput)
+	case "EventBridge":
+		resourceNameKey, resourceNameValue = extractRuleName(requestInput)
+	case "SFN":
+		resourceNameKey, resourceNameValue = extractStateMachineName(requestInput)
+	}
+	return resourceNameKey, resourceNameValue
+}
+
+func extractQueueName(requestInput middleware.InitializeInput) (resourceNameKey string, resourceNameValue string) {
```

Comment:
I'm not a fan of named return values, but if you are going to use them, the names should be used within the function. Having the name be `resourceNameKey` and returning the value of `tagQueueName` is confusing, as is `queueNameValue` and `resourceNameValue`.

Reply:
updated :), thank you!